exp/api/v1alpha4/awsmanagedmachinepool_types.go

```diff
@@ -43,6 +43,16 @@ const (
 	Al2Arm64 ManagedMachineAMIType = "AL2_ARM_64"
 )
 
+// ManagedMachinePoolCapacityType specifies the capacity type to be used for the managed MachinePool.
+type ManagedMachinePoolCapacityType string
+
+const (
+	// ManagedMachinePoolCapacityTypeOnDemand is the default capacity type, to launch on-demand instances.
+	ManagedMachinePoolCapacityTypeOnDemand ManagedMachinePoolCapacityType = "ON_DEMAND"
+	// ManagedMachinePoolCapacityTypeSpot is the spot instance capacity type to launch spot instances.
+	ManagedMachinePoolCapacityTypeSpot ManagedMachinePoolCapacityType = "SPOT"
```

Comment:
Could we make these enums a bit more Kubernetes friendly,
i.e. `onDemand` and `spot`, and then convert to the AWS (Java) backend constants in the controller?

Reply:
Ack